src/super_gradients/training/pre_launch_callbacks/pre_launch_callbacks.py

```diff
@@ -104,39 +111,51 @@ def __call__(self, cfg: DictConfig) -> DictConfig:
         tmp_cfg.training_hyperparams.kill_ddp_pgroup_on_end = False
         tmp_cfg.pre_launch_callbacks_list = []
 
+        fastest_batch_time = np.inf
+        fastest_batch_size = curr_batch_size
+
         while True:
             tmp_cfg.dataset_params.train_dataloader_params.batch_size = curr_batch_size
 
             try:
+                passes_start = cv2.getTickCount()
                 Trainer.train_from_config(tmp_cfg)
-
+                curr_batch_time = (cv2.getTickCount() - passes_start) / cv2.getTickFrequency()
+                logger.info(f"Batch size = {curr_batch_size} time for {self.num_forward_passes} forward passes: {curr_batch_time} seconds.")
+                if curr_batch_time < fastest_batch_time:
+                    fastest_batch_size = curr_batch_size
+                    fastest_batch_time = curr_batch_time
             except RuntimeError as e:
                 if "out of memory" in str(e):
                     if curr_batch_size == self.min_batch_size:
                         logger.error("Ran out of memory for the smallest batch, try setting smaller min_batch_size.")
                         raise e
                     else:
-                        logger.info(f"Ran out of memory for {curr_batch_size}, setting batch size to {curr_batch_size - self.size_step}.")
-                        self._adapt_lr_if_needed(cfg, found_batch_size=curr_batch_size - self.size_step)
-                        cfg.dataset_params.train_dataloader_params.batch_size = curr_batch_size - self.size_step
-                        self._clear_model_gpu_mem(model)
+                        selected_batch_size = curr_batch_size - self.size_step if self.mode == "largest" else fastest_batch_size
+                        msg = f"Ran out of memory for {curr_batch_size}, setting batch size to {selected_batch_size}."
+                        self._inject_selected_batch_size_to_config(cfg, model, msg, selected_batch_size)
                         return cfg
```

Comment:
Can we do `return self._inject_selected_batch_size_to_config(cfg, model, msg, selected_batch_size)` instead?